datasetviewer/tests/preview_presenter_test.py

```diff
@@ -0,0 +1,83 @@
+import unittest
+import mock
+
+import numpy as np
+
+from collections import OrderedDict as DataSet
+
+from datasetviewer.preview.PreviewPresenter import PreviewPresenter
+from datasetviewer.preview.interfaces.PreviewView import PreviewView
+from datasetviewer.mainview.MainViewPresenter import MainViewPresenter
+from datasetviewer.dataset.Variable import Variable
+
+class PreviewPresenterTest(unittest.TestCase):
+
+    def setUp(self):
+
+        self.view = mock.create_autospec(PreviewView)
+
+        self.master = mock.create_autospec(MainViewPresenter)
+
+        self.var_name = "Key"
+        self.var_dims = (8, 5)
+        self.fake_data = DataSet()
+        self.fake_data[self.var_name] = Variable(self.var_name, np.random.rand(*self.var_dims))
+
+        # The fake preview text that should be generated from the fake data
+        self.fake_preview_text = self.var_name + "\n" + str(self.var_dims)
+
+    def test_presenter_throws_if_view_none(self):
+        '''
+        Ensure that the PreviewPresenter throws an exception when given a None PreviewView.
+        '''
+        with self.assertRaises(ValueError):
+            PreviewPresenter(None)
+
+    def test_create_preview_text(self):
+        '''
+        Test that the Preview text that is generated after setting the data contains the expected key and data
+        dimensions
+        '''
+        prev_presenter = PreviewPresenter(self.view)
+        prev_presenter.set_data(self.fake_data)
+        self.assertEqual(prev_presenter._create_preview_text(self.var_name), self.fake_preview_text)
+
+    def test_call_to_create_preview_text(self):
+        '''
+        Test that the PreviewPresenter calls the `_create_preview_text` method once the `_add_preview_entry` method
+        has been called.
+        '''
+        prev_presenter = PreviewPresenter(self.view)
+
+        with mock.patch('datasetviewer.preview.PreviewPresenter.PreviewPresenter._create_preview_text') as prev_text:
+            prev_presenter._add_preview_entry(self.var_name)
+            prev_text.assert_called_once()
+
+    def test_create_preview_calls_add_to_list(self):
+        '''
+        Test that the PreviewPresenter calls the PreviewView's `_add_entry_to_list` method once with the expected text
+        once the data attribute has been set in the PreviewPresenter.
+        '''
+        prev_presenter = PreviewPresenter(self.view)
+        prev_presenter.set_data(self.fake_data)
+        self.view.add_entry_to_list.assert_called_once_with(self.fake_preview_text)
+
+    def test_call_to_add_preview_entry(self):
+        '''
+        Test that the `_add_preview_entry` method has been called once with the expected text once the data attribute
+        has been set.
+        '''
+        prev_presenter = PreviewPresenter(self.view)
+        prev_presenter.set_data(self.fake_data)
+
+        with mock.patch('datasetviewer.preview.PreviewPresenter.PreviewPresenter._add_preview_entry') as add_prev:
```

Comment:
I think the testing here is maybe going too far. We are now testing that internal methods are called in the right sequence. This has the danger to making the tests too fragile. i.e. I start renaming internal methods or changing internal behaviour (that does not affect dependent objects or the public interface) and the tests break.

I believe you are already testing that the views are called correctly above, so I think this test is redundant.